crates/rustfix/src/lib.rs

```diff
@@ -253,8 +253,19 @@ impl CodeFix {
 
 /// Applies multiple `suggestions` to the given `code`.
 pub fn apply_suggestions(code: &str, suggestions: &[Suggestion]) -> Result<String, Error> {
+    let mut already_applied = HashSet::new();
     let mut fix = CodeFix::new(code);
     for suggestion in suggestions.iter().rev() {
+        // This assumes that if any of the machine applicable fixes in
```

Comment:
Decided to duplicate the work in also `rustfix`. I didn't do it in `CodeFix` as it might require a longer lifetime bound than it was.